### DISCUSSION TITLE
Dynamic rendering of Components in Container

### DISCUSSION THREAD

**Discussion by pralad-p** [ArthurSonzogni/FTXUI#658] [Q&A] [answered]
I have a use-case which involves adding checkboxes (or just for an example, any `Component`) to a `Container::Vertical` dynamically. What do I mean? 
Suppose I have a menu that has a menu option that calls `fileSelectorCallback` whenever Enter is hit on an Menu option. 
```cpp
    MenuOption file_menu_option;
    file_menu_option.on_enter = fileSelectorCallback;
```
`fileSelectorCallback` is called, which looks like this:
```cpp
    auto fileSelectorCallback = [&]() {
                taskComponents.clear();
                taskComponents.push_back(checkbox);
    };
```
`taskComponents` here is the Children to the `Container::Vertical` I would like, to keep changing it's contents based on a menu option selected. It is defined as:
```cpp
Components taskComponents;
```
I'm having issues with rendering the children of the container i.e. taskComponents since putting it in a Renderer would make me lose interactivity with it. For example, this is the solution I tried as well:
```cpp
auto dynamicContentRenderer = ftxui::Renderer([&]() -> ftxui::Element {
        std::vector<ftxui::Element> elements;
        for (const auto& component : taskComponents) {
            elements.push_back(component->Render());
        }
        return ftxui::vbox(std::move(elements));
    });
```
which works, but I lose interactivity. Great library, btw!

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello @pralad-p,

I wish there was a more "functional" way of doing it, but at least with the `ftxui/component` subpart of FTXUI, the best I have to offer is an "imperative" way of doing things: Define a `Container::Vertical({})` and manually use `ComponentBase::Add` to add childrent to the tree.

If you have any API suggestion that would make your life easier, please don't hesitate to propose it.

**Reply by pralad-p** [reply]
Understood, that works for me. For now, it looks like a niche use-case but if I encounter it again, I'll make sure to drop a PR referencing this. Thanks again.
